doc/extending/using_params.txt

```diff
@@ -0,0 +1,214 @@
+    return None.
+
+
+Signature changes from having params
+------------------------------------
+
+Having declared a params for your Op will affect the expected
+signature of :meth:`perform`.  The new expected signature will have an
+extra parameter at the end which corresponds to the params object.
+
+.. warning::
+
+   If you do not account for this extra parameter, the code will fail
+   at runtime if it tries to run the python version.
+
+Also, for the C code, the `sub` dictionary will contain an extra entry
+`'params'` which will map to the variable name of the params object.
+This is true for all methods that recieve a `sub` parameter, so this
+means that you can use your params in the :meth:`c_init_code_struct
+<Op.c_init_code_struct>` method.
+
+
+A simple example
+----------------
+
+This is a simple example which uses a params object to pass a value.
+This Op will multiply a scalar input by a fixed floating point value.
+
+Since the value in this case is a python float, we chose Generic as
+the params type.
+
+.. testcode::
+
+   from theano import Op
+   from theano.gof.type import Generic
+   from theano.scalar import as_scalar
+
+   class MulOp(Op):
+       params_type = Generic()
+       __props__ = ('mul',)
+
+      def __init__(self, mul):
+          self.mul = float(mul)
+
+      def get_params(self, node):
+          return self.mul
+
+      def make_node(self, inp):
+          inp = as_scalar(inp)
+          return Apply(self, [inp], [inp.type()]
+
+      def perform(self, node, inputs, output_storage, params):
+          # Here params is a python float so this is ok
+          output_storage[0][0] = inputs[0] * params
+
+      def c_code(self, node, name, inputs, outputs, sub):
+          return ("%(z)s = %(x)s * PyFloat_AsDouble(%(p)s);" %
+                  dict(z=outputs[0], x=inputs[0], p=sub['params']))
+
+.. testoutput::
+   :hide:
+
+
+A more complex example
+----------------------
+
+This is a more complex example which actually passes multiple values.  It does a linear combination of two values using floating point weights.
+
+.. testcode::
+
+   from theano import Op
+   from theano.gof.type import Generic
+   from theano.scalar import as_scalar
+
+   class ab(object):
+       def __init__(self, alpha, beta):
+           self.alpha = alpha
+           self.beta = beta
+
+
+   class Mix(Op):
+       params_type = Generic()
+       __props__ = ('alpha', 'beta')
+
+       def __init__(self, alpha, beta):
+           self.alpha = alpha
+           self.beta = beta
+
+       def get_params(self, node):
+           return ab(alpha=self.alpha, beta=self.beta)
+
+       def make_node(self, x, y):
+           x = as_scalar(x)
+           y = as_scalar(y)
+           return Apply(self, [x, y], [x.type()]
+
+       def c_code(self, node, name, inputs, outputs, sub):
+           return """
+   PyObject *tmp;
+   double a, b;
+   tmp = PyObject_GetAttrString(%(p)s, "alpha");
+   if (tmp == NULL)
+     %(fail)s
```

Comment:
put {} around fail. We don't know in the futur if it will change and contain multiple statement. So we are better to give good example.